src/libraries/src.proj

```diff
@@ -2,41 +2,48 @@
 
   <PropertyGroup>
     <TraversalGlobalProperties>BuildAllProjects=true</TraversalGlobalProperties>
-    <NativeBinPlaceDependsOnTargets Condition="'$(TargetOS)' == 'Browser'">BuildWasmRuntimes</NativeBinPlaceDependsOnTargets>
   </PropertyGroup>
 
   <ItemGroup>
-    <NativeBinPlaceProject Include="Native\native-binplace.proj" />
+    <_allSrc Include="$(MSBuildThisFileDirectory)*\src\*.csproj"
+             Exclude="@(ProjectExclusions)" />
+    <NonNetCoreAppProject Include="@(_allSrc)"
+                          Exclude="@(NetCoreAppLibrary->'%(Identity)\src\%(Identity).csproj')" />
+    <NetCoreAppProject Include="$(CoreLibProject);
+                                @(_allSrc);
+                                $(MSBuildThisFileDirectory)Microsoft.VisualBasic.Core\src\Microsoft.VisualBasic.Core.vbproj;
+                                $(MSBuildThisFileDirectory)System.Runtime.CompilerServices.Unsafe\src\System.Runtime.CompilerServices.Unsafe.ilproj"
+                       Exclude="@(NonNetCoreAppProject)" />
     <ManualShimProject Include="shims\manual\*.csproj" />
     <ApiCompatProject Include="shims\ApiCompat.proj"
-                      Condition="'$(DotNetBuildFromSource)' != 'true' and '$(BuildTargetFramework)' == '$(NetCoreAppCurrent)'" />
-  </ItemGroup>
-
-  <!-- Restore only and build before/after. -->
-  <ItemGroup Condition="'$(MSBuildRestoreSessionId)' != ''">
-    <ProjectReference Include="@(NativeBinPlaceProject)" />
-    <ProjectReference Include="@(ManualShimProject)" />
-    <ProjectReference Include="@(ApiCompatProject)" />
+                      Condition="'$(DotNetBuildFromSource)' != 'true' and
+                                 '$(BuildingNETCoreAppVertical)' == 'true' and
+                                 '$(RunApiCompat)' != 'false'" />
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="$(MSBuildThisFileDirectory)*\src\*.csproj" Exclude="@(ProjectExclusions)" />
-    <ProjectReference Include="$(MSBuildThisFileDirectory)*\src\*.ilproj" Exclude="@(ProjectExclusions)" />
-    <ProjectReference Include="$(MSBuildThisFileDirectory)*\src\*.vbproj" Exclude="@(ProjectExclusions)" />
-  </ItemGroup>
+    <ProjectReference Include="@(NetCoreAppProject)" />
 
-  <Import Condition="'$(TargetOS)' == 'Browser'" Project="$(MonoProjectRoot)wasm\wasm.targets" />
+    <!-- Restore only and build before/after. -->
+    <ProjectReference Include="@(NonNetCoreAppProject);
+                               @(ManualShimProject);
+                               @(ApiCompatProject)"
+                      Condition="'$(MSBuildRestoreSessionId)' != ''" />
+  </ItemGroup>
 
-  <Target Name="NativeBinPlace"
```

Comment:
https://github.com/dotnet/runtime/blob/master/docs/workflow/building/libraries/webassembly-instructions.md#building-the-webassembly-runtime-files -- this task is not unused

Reply:
I'm not removing it, only moving it to where it belongs. Native binplace is now invoked after the native build in build-native.proj and the corresponding wasm files are moved into wasm.targets.

Reply:
I didn't know that this was documented
 I'll send a PR to update the doc.

Reply:
https://github.com/dotnet/runtime/pull/39891